cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -287,7 +278,13 @@ public void executeReportGeneration(HttpServiceRequest request, HttpServiceRespo
       }
       LOG.debug("Wrote to startFile {}", startFile.toURI());
       // Generate the report asynchronously
-      getReportGenerationExecutor().execute(() -> tryGenerateReport(reportRequest, reportIdDir, reportId));
+      REPORT_EXECUTOR.execute(() -> {
+        // directly return without generating the report if the executor is being shutdown
```

Comment:
Why still run it?

Reply:
do you mean this part  ```
        if (Thread.currentThread().isInterrupted()) {
          return;
        }``` should be moved out of the execute?

Reply:
if the executor is shutdown,  the execution daemon threads are anyway going to be shut down, i am not sure if i understand the purpose of this comment and the interrupted check below.